File: Q1496 - Path Crossing/1496.cpp
class Solution {
public:
    bool isPathCrossing(string path) {

        int x = 0, y = 0;
        set<pair<int, int>> points;
        points.insert({ x,y });

        for (int i = 0; i < path.length(); i++) {
            if (path[i] == 'N')
            {
                y++;
            }
            if (path[i] == 'S')
            {
                y--;
            }
            if (path[i] == 'E')
            {
                x++;
            }
            if (path[i] == 'W')
            {
                x--;
            }
            auto temp = points.insert({ x,y });
            if (temp.second == false)
                return true;
        }
        return false;
    }
};

File: Q1496 - Path Crossing/1496.py
class Solution:
    def isPathCrossing(self, path: str) -> bool:
        x=0
        y=0
        points= set()

        str="{}, {}".format(x,y)
        points.add(str)
        for i in range(len(path)):
            if(path[i]=='N'):
                y=y+1

            if(path[i]=='S'):
                y=y-1

            if(path[i]=='E'):
                x=x+1

            if(path[i]=='W'):
                x=x-1

            l = len(points) 
            str="{}, {}".format(x,y)
            points.add(str)

            if (len(points) == l):
                return True

        return False